### PR TITLE
[PoC] Alternative to MetricProducer

### PR DESCRIPTION
**Pull request by dashpole**
This is an attempt at creating an OpenCensus bridge without using the metric.Producer interface.

For push, this is straightforward.  I provide an "exporter wrapper" that adds OpenCensus metrics in.

For pull, I provide a "manual reader wrapper".  The tricky part is that the user needs to override the prometheus exporter's reader to use it.  Overall, I think it will be slightly confusing ("why do I need to set this Reader field on my Prometheus exporter"), but it isn't too bad of a solution for a bridge.  I'm relatively pleased with the current state and UX.

The primary drawback of this approach is that it requires making it possible to replace the Reader on the prometheus exporter to work.  This means that someone could, in theory, override the Reader with one that uses delta temporality.

### CODE REVIEW

**Comment by MrAlias** [comment]
> Then I make a "OpenCensusBridgePrometheusExporter" monstrocity that overrides _just_ the Collect function to add in OpenCensus metrics.

😆

File: bridge/opencensus/metric.go [hunk 1]
```diff
@@ -18,81 +18,121 @@ import (
...
+}
+
+func NewBridgedPrometheusExporter(opts ...prometheus.Option) (*prometheus.Exporter, error) {
+	exporter, err := prometheus.New(opts...)
 	if err != nil {
-		otel.Handle(err)
+		return nil, err
+	}
+	exporter.Reader = &PrometheusExporter{
+		Reader:  metric.NewManualReader(),
```
Comment (dashpole) [inline bridge/opencensus/metric.go:99]: TODO: make sure we can access the aggregation selector passed to the prometheus exporter in options.

Comment (dashpole) [reply bridge/opencensus/metric.go]: done

**Comment by dashpole** [comment]
I refactored the pull exporter bridge, and i'm now relatively pleased with the implementation.

**Comment by dashpole** [comment]
Closing based on https://github.com/open-telemetry/opentelemetry-specification/issues/3599#issuecomment-1660583482

**Pull request closed without merging**